song-server/src/main/java/bio/overture/song/server/security/StudySecurity.java

```diff
@@ -36,6 +38,19 @@
   @NonNull private final String systemScope;
 
   public boolean authorize(@NonNull Authentication authentication, @NonNull final String studyId) {
+    if (authentication instanceof ExpiringOauth2Authentication) {
+      val auth = (ExpiringOauth2Authentication) authentication;
+
+      if (auth.getExpiry() == 0 ) {
+        log.info("Expired token detected; authorization denied.");
+        return false;
+      }
+      log.info("Token is not expired; authorization continues.");
+    } else {
+      log.error("Unknown authentication type detected!");
+      return false;
+    }
+
```

Comment:
Please put this in some static method, and call it in both the `StudySecurity.authorize()` and `SystemSecurity.authorize()` methods. Endpoints such as [analysisType.register](https://github.com/overture-stack/SONG/blob/develop/song-server/src/main/java/bio/overture/song/server/controller/analysisType/AnalysisTypeController.java#L106-L109) require systemSecurity only.